lib/pymedphys/_mosaiq/delivery.py

```diff
@@ -281,36 +286,61 @@ def delivery_data_sql(connection, field_id):
             TxFieldPoint.Coll_Ang,
             TxFieldPoint.Coll_Y1,
             TxFieldPoint.Coll_Y2
+            {',TxFieldPoint.RowVers' if include_rowvers else ''}
         FROM TxFieldPoint
         WHERE
             TxFieldPoint.FLD_ID = %(field_id)s
+        ORDER BY
+            TxFieldPoint.Point
         """,
             {"field_id": field_id},
         )
     )
 
+    if include_rowvers:
+        txfield_results[-1] = struct.unpack("Q", txfield_results[-1])
+        txfieldpoint_results[-1] = struct.unpack("Q", txfieldpoint_results[-1])
+
     return txfield_results, txfieldpoint_results
 
 
 def fetch_and_verify_mosaiq_sql(connection, field_id):
-    reference_results = delivery_data_sql(connection, field_id)
-    test_results = delivery_data_sql(connection, field_id)
+    reference_txfield_results, reference_txfieldpoint_results = delivery_data_sql(
+        connection, field_id, include_rowvers=True
+    )
+    test_txfield_results, test_txfieldpoint_results = delivery_data_sql(
+        connection, field_id, include_rowvers=True
+    )
 
     agreement = False
 
     while not agreement:
-        agreements = []
-        for ref, test in zip(reference_results, test_results):
+        agreements = [np.all(reference_txfield_results == test_txfield_results)]
+        for ref, test in zip(reference_txfieldpoint_results, test_txfieldpoint_results):
             agreements.append(np.all(ref == test))
 
         agreement = np.all(agreements)
         if not agreement:
             print("Mosaiq sql query gave conflicting data.")
+
+            # log mismatched values output
+            logger = logging.getLogger("mosaiq_delivery_data_sql")
+            logger.error("Mismatch of delivery_data_sql results:")
+
+            logger.error(pformat(reference_txfield_results))
+            logger.error(pformat(test_txfield_results))
+            logger.error(pformat(reference_txfieldpoint_results))
+            logger.error(pformat(test_txfieldpoint_results))
+
             print("Trying again...")
-            reference_results = test_results
-            test_results = delivery_data_sql(connection, field_id)
+            reference_txfield_results = test_txfield_results
+            reference_txfieldpoint_results = test_txfieldpoint_results
+            test_txfield_results, test_txfieldpoint_results = delivery_data_sql(
+                connection, field_id, include_rowvers=True
+            )
 
-    return test_results
+    # TODO: remove rowverse from test_results
```

Comment:
should the rowversion be removed before returning?

Reply:
The only place this function here is utilised is within the following class method (which is within this module):

https://github.com/pymedphys/pymedphys/blob/fdb7fd1781416b3d31a69fb4af5af50933beb1c3/lib/pymedphys/_mosaiq/delivery.py#L384-L420

Everything in that function is using positive integer indexing so permanently allowing a new item to be included at the end shouldn't interfere. I think, remove all branching logic regarding `RowVers` and just include it all through.

---

Also, of note, the public API where all this code here is exposed is via:

```python
import pymedphys

delivery = pymedphys.Delivery.from_mosaiq(connection, field_id)
```

So as long as the output of that remains constant, we have quite a bit of flexibility within these internal functions here.